qa/tasks/cephadm.py

```diff
@@ -983,7 +983,22 @@ def initialize_config(ctx, config):
     if not hasattr(ctx, 'daemons'):
         first_ceph_cluster = True
 
-    # cephadm mode?
+     # image
+    ctx.ceph[cluster_name].image = config.get('image')
+    ref = None
+    if not ctx.ceph[cluster_name].image:
+        sha1 = config.get('sha1')
+        if sha1:
+            ctx.ceph[cluster_name].image = 'quay.io/ceph-ci/ceph:%s' % sha1
+            ref = sha1
+        else:
+            # hmm, fall back to branch?
+            branch = config.get('branch', 'master')
+            ref = branch
+            ctx.ceph[cluster_name].image = 'quay.io/ceph-ci/ceph:%s' % branch
+    log.info('Cluster image is %s' % ctx.ceph[cluster_name].image)
```

Comment:
can you convert this into a function?

Reply:
hmm actually `ref` variable is used in `download_cephadm` so I think I should better move it back and just add an `if is istance` to check if it's there cause otherwise a lot of refactoring needs to be done.

Reply:
fair enough. (Still, I think it would be possible to return `ref` from that new function)

Reply:
yeah I will do it this way, for some reason I thought that it's used in another fucntion as well and it would be weird to call it multiple times just to get the`ref`